org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -118,9 +123,27 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
 
     try {
       // Load files to memory and create and invoke the code execution environment
-      CodeBuilderWrapper codeBuilderWrapper =
-          new CodeBuilderWrapper(userProjectFileLoader, outputAdapter);
-      codeBuilderWrapper.executeCodeBuilder(executionType, compileList);
+      CodeBuilderRunnable codeBuilderRunnable =
+          new CodeBuilderRunnable(userProjectFileLoader, outputAdapter, executionType, compileList);
+      Thread codeBuilderThread = new Thread(codeBuilderRunnable);
+      // start code build and execute in a thread
+      codeBuilderThread.start();
+      while (codeBuilderThread.isAlive()) {
+        // sleep for CHECK_THREAD_INTERVAL_MS, then check if we've lost the connection to the
+        // input or output adapter. This means we have lost connection to the end user (either
+        // because
+        // they terminated their program or some other issue), and we should stop executing their
+        // code.
+        Thread.sleep(CHECK_THREAD_INTERVAL_MS);
+        if (codeBuilderThread.isAlive()
+            && (!inputAdapter.hasActiveConnection() || !outputAdapter.hasActiveConnection())) {
```

Comment:
What would be a case where the inputAdapter would be inactive but the outputAdapter would be active? Is it worth always making both requests, or is it sufficient to ping only one service?

Reply:
If we had some sort of aws issue where sqs was not available or the websocket wasn't available only one could be down.